### PR TITLE
Fixed a crash when loading empty video files

### LINKED ISSUE #19124: Game crashed every time, when i try to watch Mission Brief in Dune 2000
## Issue Summary
I installed additional content in Dune 2000, and when I try to watch Mission Brief, the game crashed every time. But other games OpenRA Red Alert & OpenRA - Tiberian Down work perfectly, with briefings videos. 

DESCRIPTION

## Reproduction

I opened the game Dune 2000.
Click on Single player
Click on Mission
Chose any mission 
Click Watch Briefing

I am on MacBook Pro 13 with Apple M1

and every time game crashed

## Expected behavior

I expect to play the Mission Brief

## Screenshots / Screen recordings / Replays

Here is installed Campaign Briefings from CD.
Screenshot 2021-02-03 at 16 29 57

Here is my exceptional file from crash
[exception-2021-02-03T153128Z.log](https://github.com/OpenRA/OpenRA/files/5919523/exception-2021-02-03T153128Z.log)

And here is warning
Screenshot 2021-02-03 at 16 31 58

### PR DESCRIPTION
**Pull request by Mailaender**
Closes https://github.com/OpenRA/OpenRA/issues/19124

### CODE REVIEW

File: OpenRA.Mods.Common/Widgets/VideoPlayerWidget.cs [hunk 1]
```diff
@@ -46,7 +46,14 @@ public void Load(string filename)
 			if (filename == cachedVideoFileName)
 				return;
 
-			var video = VideoLoader.GetVideo(Game.ModData.DefaultFileSystem.Open(filename), true, Game.ModData.VideoLoaders);
+			var stream = Game.ModData.DefaultFileSystem.Open(filename);
+			if (stream.Length == 0)
```
Comment (abcdefg30) [inline OpenRA.Mods.Common/Widgets/VideoPlayerWidget.cs]: Wouldn't it make sense to have this check as part of `IsWsa` and `IsWestwoodVqa`?

File: OpenRA.Mods.Common/Widgets/VideoPlayerWidget.cs [hunk 2]
```diff
@@ -46,7 +46,14 @@ public void Load(string filename)
...
+			{
+				Log.Write("debug", $"Empty dummy file {filename} detected.");
+				return;
+			}
+
+			var video = VideoLoader.GetVideo(stream, true, Game.ModData.VideoLoaders);
 			Open(video);
```
Comment (abcdefg30) [inline OpenRA.Mods.Common/Widgets/VideoPlayerWidget.cs:51]: While we're here, can you put a null check inside of `Open`? `GetVideo` might return null.

File: OpenRA.Mods.Common/Widgets/VideoPlayerWidget.cs (hunk 1 above)
Comment (Mailaender) [reply OpenRA.Mods.Common/Widgets/VideoPlayerWidget.cs]: No, those functions don't have the filename for logging.

File: OpenRA.Mods.Common/Widgets/VideoPlayerWidget.cs (hunk 2 above)
Comment (Mailaender) [reply OpenRA.Mods.Common/Widgets/VideoPlayerWidget.cs:51]: Added.

File: OpenRA.Mods.Common/Widgets/VideoPlayerWidget.cs (hunk 1 above)
Comment (abcdefg30) [reply OpenRA.Mods.Common/Widgets/VideoPlayerWidget.cs]: `VideoLoader.GetVideo` would return null which means we could still log here.

Comment (Mailaender) [reply OpenRA.Mods.Common/Widgets/VideoPlayerWidget.cs]: Okay done as that would cover more crashes.

File: OpenRA.Mods.Common/Widgets/VideoPlayerWidget.cs [hunk 3]
```diff
@@ -46,7 +46,14 @@ public void Load(string filename)
...
 				return;
 
-			var video = VideoLoader.GetVideo(Game.ModData.DefaultFileSystem.Open(filename), true, Game.ModData.VideoLoaders);
+			var stream = Game.ModData.DefaultFileSystem.Open(filename);
+			var video = VideoLoader.GetVideo(stream, true, Game.ModData.VideoLoaders);
+			if (video == null)
+			{
+				Log.Write("debug", $"Invalid video file {filename} detected.");
+				return;
+			}
```
Comment (abcdefg30) [inline OpenRA.Mods.Common/Widgets/VideoPlayerWidget.cs]: Can we move this into `Open` to cover the videos played via Lua as well?

Comment (Mailaender) [reply OpenRA.Mods.Common/Widgets/VideoPlayerWidget.cs]: That kills the logging as we have no filename but whatever.

Comment (abcdefg30) [reply OpenRA.Mods.Common/Widgets/VideoPlayerWidget.cs]: Oh right. Hm, if you really think we need the logging we could revert this, as the Lua call seems to be wrapped in a try-catch. But I'm not sure if it's worth it. 🤔

**Review by abcdefg30 — COMMENTED** [review]
Works as advertised. It does look a bit weird though without feedback to the user:
grafik
Can we add a label that the video loading failed? The label can be set up to be only visible when `video` is null, so it shouldn't require invasive changes to the VideoPlayerWidget.

**Comment by Mailaender** [comment]
No, that is not required in my opinion.

**Review by penev92 — DISMISSED** [review]
👍 

I guess our `Rebase and merge` strategy will remove the merge commit, so squashing is a matter of taste.

**Comment by abcdefg30** [comment]
> so squashing is a matter of taste.

I took the liberty to rebase and squash.

**Review by abcdefg30 — APPROVED** [review]

**Comment by abcdefg30** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/d31010b52e800ba4cd4b5085e989c543b30ff001...8d968c292f84280c72ff57e98a2d748c9c94566f)

**Pull request merged**